### PR TITLE
Feature/wandb

### PR DESCRIPTION
**Pull request by asbharath**
- Added wandb basic functionality to train.py

### CODE REVIEW

File: dockerfiles/tensorflow.dockerfile [hunk 1]
```diff
@@ -1,14 +1,15 @@
 FROM tensorflow/tensorflow:2.3.0-gpu
 
 RUN apt-get update && \
-    apt-get install -y libsm6 libxrender1 libxext6 libgl1-mesa-glx && \
+    apt-get install -y libsm6 libxrender1 libxext6 libgl1-mesa-glx vim && \
```
Comment (Net-Mist) [inline dockerfiles/tensorflow.dockerfile]: This docker image should stay as light as possible, and vim shouldn't be specified for all user.
If someone wants to use vim, he should create a new dockerfile for his own purpose

File: dockerfiles/upstride.dockerfile [hunk 2]
```diff
@@ -1,14 +1,15 @@
 FROM eu.gcr.io/fluid-door-230710/upstride:py-1.1.1-tf2.3.0-gpu
 
 RUN apt-get update && \
-    apt-get install -y libsm6 libxrender1 libxext6 libgl1-mesa-glx && \
+    apt-get install -y libsm6 libxrender1 libxext6 libgl1-mesa-glx vim && \
```
Comment (Net-Mist) [inline dockerfiles/upstride.dockerfile]: same remark as above

File: train.py [hunk 3]
```diff
@@ -24,6 +24,9 @@
     [int, 'n_layers_before_tf', 0, 'when using mix framework, number of layer defined using upstride', lambda x: x >= 0],
     [str, 'load_searched_arch', '', 'model definition file containing the searched architecture'],
     [str, "model_name", '', 'Specify the name of the model', lambda x: x in model_name_to_class],
+    [bool, "use_wandb", False, 'enable if we want to utilize weights and biases'],
+    [str, 'project', 'project0', 'Unique project name within which the training runs are executed in wandb',],
+    [str, 'run_name', '', 'Unique run name within which the training runs are executed in wandb',],
```
Comment (Net-Mist) [inline train.py]: these 3 new parameters should be put in a namespace to keep a clean organization

File: train.py [hunk 4]
```diff
@@ -37,6 +40,10 @@ def main():
   """
   args = argparse.parse_cmd(arguments)
   args['server'] = alchemy_api.start_training(args['server'])
+  if args['use_wandb'] and "tensorflow" in args['framework']:
```
Comment (Net-Mist) [inline train.py]: add a comment in the code to explain why there is a condition on tensorflow

**Review by Net-Mist — CHANGES_REQUESTED** [review]
There are a few points to correct before merging. See comments in the code

File: dockerfiles/upstride.dockerfile (hunk 2 above)
Comment (asbharath) [reply dockerfiles/upstride.dockerfile]: agreed. Updated the same

File: train.py (hunk 3 above)
Comment (asbharath) [reply train.py]: agreed. Updated the same

File: train.py (hunk 4 above)
Comment (asbharath) [reply train.py]: Let me know if this comment is okie. or should we be more explicit as to why we don't report Upstride.

File: train.py (hunk 3 above)
Comment (Net-Mist) [reply train.py]: I think you defined the namespace but you forgot to update the code when you call the argument

Comment (asbharath) [reply train.py]: Its fixed now and tested. Sorry

**Pull request merged**